cabal-install/Distribution/Client/SetupWrapper.hs

```diff
@@ -414,7 +420,8 @@ internalSetupMethod verbosity options bt args = do
   inDir (useWorkingDir options) $ do
     withEnv "HASKELL_DIST_DIR" (useDistPref options) $
       withExtraPathEnv (useExtraPathEnv options) $
-        buildTypeAction bt args
+        withEnvOverrides (useExtraEnvOverrides options) $
```

Comment:
Having to explicitly set environment variables here feels somewhat strange (see https://github.com/haskell/cabal/issues/5164#issuecomment-372447822). But then again, we also do this for `HASKELL_DIST_DIR` and `PATH` above, so I opted to do the same here for consistency's sake.